crates/rooch-framework/sources/auth_validator/ecdsa_k1_validator.move

```diff
@@ -83,61 +82,95 @@ module rooch_framework::ecdsa_k1_validator {
-        // TODO handle non-ed25519 auth key and address relationship
-        // let auth_key = ecdsa_k1_authentication_key(&authenticator_payload);
-        // let auth_key_in_account = get_authentication_key(ctx, storage_context::sender(ctx));
-        // assert!(
-        //    auth_key_in_account == auth_key,
-        //    auth_validator::error_invalid_account_auth_key()
-        // );
+    /// The authentication key exists in account or not.
+    public fun is_authentication_key_in_account(ctx: &StorageContext, addr: address): bool {
+        option::is_some(&get_authentication_key_option_from_account(ctx, addr))
+    }
+
+    /// Get the authentication key of the given account.
+    public fun get_authentication_key_from_account(ctx: &StorageContext, addr: address): vector<u8> {
+        assert!(is_authentication_key_in_account(ctx, addr), error::not_found(EAuthenticationKeyNotFoundInAccount));
+        option::extract(&mut get_authentication_key_option_from_account(ctx, addr))
+    }
+
+    /// Only validate the authenticator's signature.
+    public fun validate_signature(authenticator_payload: &vector<u8>, tx_hash: &vector<u8>) {
         assert!(
             ecdsa_k1::verify(
-                &ecdsa_k1_signature(&authenticator_payload),
-                &ecdsa_k1_public_key(&authenticator_payload),
-                &storage_context::tx_hash(ctx),
-                SHA256, // KECCAK256:0, SHA256:1, TODO: The hash type may need to be passed through the authenticator
+                &get_signature_from_authenticator_payload(authenticator_payload),
+                &get_public_key_from_authenticator_payload(authenticator_payload),
+                tx_hash,
+                SHA256
             ),
             auth_validator::error_invalid_authenticator()
         );
     }
 
+    public fun validate(ctx: &StorageContext, authenticator_payload: vector<u8>) {
+        let tx_hash = storage_context::tx_hash(ctx);
+        validate_signature(&authenticator_payload, &tx_hash);
+
+        let auth_key_from_authenticator_payload = get_authentication_key_from_authenticator_payload(&authenticator_payload);
+        std::debug::print(&auth_key_from_authenticator_payload);
+        // Although we have checked public key length in rotate_authentication_key_entry function,
+        // it needs to validate the authentication key isn't empty or malformed.
+        assert!(
+           vector::length(&auth_key_from_authenticator_payload) == V_AUTHENTICATION_KEY_LENGTH,
+           error::invalid_argument(EInvalidAuthenticatorPayloadAuthKeyLength)
+        );
+    }
+
     fun pre_execute(
         _ctx: &mut StorageContext,
     ) {}
 
     fun post_execute(
-        _ctx: &mut StorageContext,
-    ) {}
+        ctx: &mut StorageContext,
+    ) {
+        let account_addr = storage_context::sender(ctx);
+        let auth_key_in_account = get_authentication_key_from_account(ctx, account_addr);
```

Comment:
Same as above, we cant use MultiChainAddress to rotate_authentication_key_entry

Reply:
You mean MultiChainAddress a Rooch address? I'll refactor `public_key_to_authentication_key` function to convert to a ETH or BTC address, TBD, and pass it in `rotate_authentication_key_entry ` to update an ETH or BTC address as an auth key.

Reply:
MultiChainAddress for ETH address when the chain is ETH.